task/test-tests-scan

```diff
@@ -0,0 +1,179 @@
+
+import json
+import os
+import shutil
+import subprocess
+import tempfile
+import unittest
+
+from lib.constants import BOTS_DIR
+from task.test_mock_server import MockHandler, MockServer
+
+ADDRESS = ("127.0.0.7", 9898)
+
+
+GITHUB_DATA = {
+    "/repos/project/repo": {
+        "default_branch": "main"
+    },
+    "/repos/project/repo/issues/3333": {
+        "title": "The issue title",
+        "body": "Some bug\n - [ ] image-refresh foonux\n",
+    },
+    "/repos/project/repo/pulls/1": {
+        "title": "PR title",
+        "number": 1,
+        "body": "This is the body",
+        "base": {"ref": "abcdef"},
+        "head": {"sha": "abcdef", "user": {"login": "cockpit-project"}},
+        "labels": [],
+    },
+    "/repos/project/repo/commits/abcdef/status?page=1&per_page=100": {
+        "state": "pending",
+        "statuses": [],
+        "sha": "abcdef",
+    },
+    "/users/user/repos": [{"full_name": "project/repo"}]
+}
+
+
+class Handler(MockHandler):
+    def do_GET(self):
+        if self.path in self.server.data:
+            self.replyJson(self.server.data[self.path])
+        elif self.path.startswith('/repos/project/repo/pulls/2'):
+            self.replyJson([])
+        elif self.path.startswith('/repos/project/repo/pulls'):
+            self.replyJson([self.server.data['/repos/project/repo/pulls/1']])
+        else:
+            self.send_error(404, 'Mock Not Found: ' + self.path)
+
+    def do_POST(self):
+        self.send_error(405, 'Method not allowed: ' + self.path)
+
+
+class TestTestsScan(unittest.TestCase):
+    def setUp(self):
+        self.temp = tempfile.mkdtemp()
+        self.server = MockServer(ADDRESS, Handler, GITHUB_DATA)
+        self.server.start()
+        self.repo = "project/repo"
+        self.pull_number = "1"
+        self.context = "fedora/nightly"
+        self.revision = "abcdef"
+
+    def tearDown(self):
+        self.server.kill()
+        shutil.rmtree(self.temp)
+
+    def run_tests_scan(self, args):
+        script = os.path.join(BOTS_DIR, "tests-scan")
+        print([script, *args])
+        proc = subprocess.Popen([script, *args], stdout=subprocess.PIPE, universal_newlines=True,
+                                env={"GITHUB_API": f"http://{ADDRESS[0]}:{ADDRESS[1]}"})
+        output, stderr = proc.communicate()
+        return proc, output, stderr
+
+    def expected_command(self):
+        expected_output = f"""./s3-streamer --repo {self.repo} --test-name pull-{self.pull_number}-\\d+-\\d+
+--github-context {self.context} --revision {self.revision} -- /bin/sh -c "PRIORITY=0005 ./make-checkout --verbose
+--repo={self.repo} {self.revision} && cd make-checkout-workdir && TEST_OS=fedora BASE_BRANCH={self.revision}
+COCKPIT_BOTS_REF=main TEST_SCENARIO=nightly ../tests-invoke --pull-number {self.pull_number}
+--revision {self.revision} --repo {self.repo}"
+"""
+        return " ".join(expected_output.splitlines()).strip()
```

Comment:
Maybe better to use this style:
```python
expected_output = ("first "
                   "second")
```
then you can avoid this split/strip/join hack